### ISSUE TITLE
`SemanticModel.GetConstantValue()` should return the name of the variable when called on a `nameof(var)` expression even when the type of the variable cannot be resolved

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [dotnet/roslyn#54082]
**Version Used**:  Visual Studio Version 16.9.4 

**TL;DR**:

When the SyntaxNode which gets passed to `GetConstantValue()` is a `nameof()` invocation expression where the argument of the `nameof()` call is a variable with **an unknown type**, `GetConstantValue()` returns an Optional with an empty string value inside (and `HasValue` set to True).

This scenario can happen during development inside an IDE (when the build can be broken while the developer is typing, or when for some reason the type hasn't been resolved).

**Steps to Reproduce**:

It's easier to write the code:

```
        public void Method(UnknownType second)
        {
            new Foo(nameof(second));

            // in the analyzer: 
            // ObjectCreationExpressionSyntax node = (ObjectCreationExpressionSyntax)c.Node;
            // var firstArgExpression = node.ArgumentList.Arguments[0].Expression;
            // var value = c.SemanticModel.GetConstantValue(firstArgExpression);
 
            // EXPECTED: value.HasValue is True; value.Value is "second"
            // ACTUAL: value.HasValue is True; value.Value is "" (empty string"
        }
```

I created a reproducer project in https://github.com/andrei-epure-sonarsource/roslyn-getconstantvalue-reproducer
- it contains a [small analyzer](https://github.com/andrei-epure-sonarsource/roslyn-getconstantvalue-reproducer/blob/main/Analyzer/AnalyzerAnalyzer.cs#L32) that calls `c.SemanticModel.GetConstantValue(firstArgExpression);` and outputs what the returned `Optional` value contains (`HasValue` and `Value`)
- it also contains a sample project so that you can see the behavior of the `GetConstantValue()` API [in various scenarios](https://github.com/andrei-epure-sonarsource/roslyn-getconstantvalue-reproducer/blob/main/Sample/Program.cs#L22)

The instructions to run locally are in the repository README file.

**Expected Behavior**:

When the type of a variable passed to `nameof()` is unknown, `SemanticModel.GetConstantValue()` for the `nameof()` expression should return an Optional that has:
- either HasValue=False
- or HasValue=True and Value= the name of the variable inside the nameof() call

**Actual Behavior**:

The Optional returned by `SemanticModel.GetConstantValue()` contains `HasValue=True` and `Value=''` (empty string).

Note: this problem has initially been reported in https://github.com/SonarSource/sonar-dotnet/issues/4180

**Comment by Youssef1313** [comment]
Most likely this was fixed in https://github.com/dotnet/roslyn/pull/52016.

**Comment by andrei-epure-sonarsource** [comment]
I updated VS to 16.10.1 and I confirm this got fixed - thus I am closing the ticket.

Indeed, it would be nice to have a regression test for this scenario.

Thanks!
